src/Infrastructure/OpenApi/Startup.cs

```diff
@@ -91,7 +91,14 @@ internal static IServiceCollection AddOpenApiDocumentation(this IServiceCollecti
             });
             services.AddScoped<FluentValidationSchemaProcessor>();
         }
+      
+        services.AddScoped<FluentValidationSchemaProcessor>(provider =>
+        {
+            var validationRules = provider.GetService<IEnumerable<FluentValidationRule>>();
+            var loggerFactory = provider.GetService<ILoggerFactory>();
 
+            return new FluentValidationSchemaProcessor(provider, validationRules, loggerFactory);
+        });
```

Comment:
The FluentValidationSchemaProcessor is already being added as a scoped service, right there, 2 lines above where you add it again... also, it's not necessary to take the dependent services out of the provider. aspnetcore DI does that automatically for you upon requesting a service.

Reply:
Ok, I understand you do the provider thing and call a specific constructor to get rid of the exception... but why don't I get that exception on my end, even though there are also 2 constructors on FluentValidationSchemaProcessor (but one of them is marked "Obsolete")... It's probably a version thing?

Reply:
I'm using all the packages' updated version. Maybe that's why I got the exception.